src/commands/upload_file.ts

```diff
@@ -177,26 +177,33 @@ new CLICommand({
 				arweave
 			});
 
-			const uploadStats: ArDriveUploadStats[] = await Promise.all(
-				filesToUpload.map(
-					async ({ parentFolderId, wrappedEntity, destinationFileName, driveKey, drivePassword }) => {
-						driveKey ??= (await parameters.getIsPrivate())
-							? await parameters.getDriveKey({
-									driveId: await arDrive.getDriveIdForFolderId(parentFolderId),
-									drivePassword,
-									useCache: true
-							  })
-							: undefined;
-
-						return {
-							wrappedEntity,
-							driveKey,
-							destFolderId: parentFolderId,
-							destName: destinationFileName
-						};
-					}
-				)
-			);
+			const uploadStats: ArDriveUploadStats[] = await (async () => {
+				const uploadStats: ArDriveUploadStats[] = [];
+
+				for (const {
+					parentFolderId,
+					wrappedEntity,
+					destinationFileName,
+					driveKey,
+					drivePassword
+				} of filesToUpload) {
+					uploadStats.push({
+						wrappedEntity,
+						driveKey:
+							driveKey || (await parameters.getIsPrivate())
```

Comment:
Let's use the nullish coalescing operator here, as it is safer than '||'.
```suggestion
							driveKey ?? (await parameters.getIsPrivate())
```